data/json/mapgen/swamp_shack.json

```diff
@@ -112,11 +124,36 @@
         "b": "f_makeshift_bed",
         "c": "f_chair",
         "d": "f_rack",
-        "i": [ "f_null", "f_null", "f_null", "f_null", "f_null", "f_null", "f_null", "f_cattails", "f_cattails", "f_cattails", "f_cattails" ],
         "n": "f_counter",
         "s": "f_counter",
         "x": "f_crate_o"
-      }
+      },
+      "place_nested": [
+        { "chunks": [ "cattail", "null" ], "x": [ 0, 6 ], "y": [ 0, 6 ], "neighbors": { "north": "forest_water" } },
```

Comment:
Curious, what does the null mean in here?

Reply:
If the null is rolled, nothing is nested. They have equal weights. So 50% chance for cattail.
This is to pad the chances a bit, to make it somewhat more like the surrounding forest.  Doing this whole "x rolls for cattail, 50% chance" makes it more like the math in forest cattail placement than just rolling "1 to 4 cattails" and placing those.